src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -101,16 +97,11 @@ public class CodeSonarPublisher extends Recorder implements SimpleBuildStep {
     private int socketTimeoutMS = -1;
     private String projectFile;
 
-    private XmlSerializationService xmlSerializationService = null;
     private HttpService httpService = null;
     private AuthenticationService authenticationService = null;
-    private IAnalysisService analysisService = null;
-    private MetricsService metricsService = null;
-    private ProceduresService proceduresService = null;
     private HubInfoService hubInfoService = null;
+    private CodeSonarServices codeSonarCacheService = null;
```

Comment:
Is this CodeSonarServices field needed?  If yes, change the name, if no, remove it and all related code.